src/pstl_offload.cpp

```diff
@@ -61,7 +103,23 @@ __internal_free(void* __user_ptr)
         }
         else
         {
-            __get_original_free()(__user_ptr);
+            if (__dlsym_called)
+            {
+                // Delay releasing till exit of dlsym. We do not overload malloc globally,
+                // so can use it safely. Do not use new to able to use free() during
+                // __delayed_free_list releasing.
+                void* __buf = malloc(sizeof(__delayed_free_list));
+                if (!__buf)
+                {
+                    throw std::bad_alloc();
+                }
+                __delayed_free_list* __h = new(__buf) __delayed_free_list{__delayed_free, __user_ptr};
```

Comment:
I guess we can avoid placement new here:

```suggestion
                __delayed_free_list* __h = static_cast<__delayed_free_list*>(malloc(sizeof(__delayed_free_list)));
                if (!__h)
                {
                    throw std::bad_alloc();
                }
                *__h = __delayed_free_list{__delayed_free, __user_ptr};
```